runtime/v2/manager.go

```diff
@@ -592,17 +604,28 @@ func (c *configurator) RegisterService(sd *grpc.ServiceDesc, ss interface{}) {
 func (c *configurator) registerQueryHandlers(sd *grpc.ServiceDesc, ss interface{}) error {
 	for _, md := range sd.Methods {
 		// TODO(tip): what if a query is not deterministic?
-		err := registerMethod(c.stfQueryRouter, sd, md, ss)
+		requestFullName, err := registerMethod(c.stfQueryRouter, sd, md, ss)
 		if err != nil {
 			return fmt.Errorf("unable to register query handler %s: %w", md.MethodName, err)
 		}
+
+		// register gRPC query method.
+		typ := gogoproto.MessageType(requestFullName)
+		if typ == nil {
+			return fmt.Errorf("unable to find message in gogotype registry: %w", err)
+		}
+		decoderFunc := func(bytes []byte) (gogoproto.Message, error) {
+			msg := reflect.New(typ.Elem()).Interface().(gogoproto.Message)
```

Comment:
are we OK with reflection on the query path? it will be a reflect call on each query.  I'm not sure of the performance overheard, should we measure? are there any security concerns?  This would mean that v2 doesn't use the generated grpc router code at all, right?

alternatives include:
- modifying codegen to produce some decoding like what's happening [in v1](https://github.com/cosmos/cosmos-sdk/blob/d2903a9eb925469ee27d5d0f44dd055c75d89ceb/baseapp/grpcrouter.go#L124)
- find a way to for this routing layer to re-use the [grpc generated code](https://github.com/cosmos/cosmos-sdk/blob/7ee3bde8f32ed0f77e7a2c40b16847213377b98b/x/auth/types/query.pb.go#L1253-L1269)

Reply:
with the de design of handlers ideally we can move away from using the grpc generated code. I dont think its worth going and changing the generator as well.